packages/wallet/src/KeyManagement/types.ts

```diff
@@ -80,7 +81,14 @@ export interface SerializableLedgerKeyAgentData extends SerializableKeyAgentData
   communicationType: CommunicationType;
 }
 
-export type SerializableKeyAgentData = SerializableInMemoryKeyAgentData | SerializableLedgerKeyAgentData;
+export interface SerializableTrezorKeyAgentData extends SerializableKeyAgentDataBase {
+  __typename: KeyAgentType.Trezor;
```

Comment:
Don't we need to save manifest and connection type?

Reply:
@mkazlauskas there is no need for that. We need that only for `initializeTrezorTransport` which should be called on app start or before interacting with a device but just once per session otherwise, it will throw an error. So, we don't need it after that. Trezor connect is a little bit different and simpler than Ledger communication

Reply:
But if we save it we can then call it within key agent (on restoration or before signing etc.) :thinking: Otherwise user (wallet) has to treat this key agent differently from other key agents, right?

Reply:
@mkazlauskas It is a little bit different than Ledger. Even if you have e.g deviceConnection you can't call e.g. deviceConnection.getFeatures(). All operations you want to perform come from TrezorConnect e.g `TrezorConnect.getFeatures()`. In the Ledger key agent, we need a device connection to prove and to keep an open channel between the end app and Cardano App on the device. Everything is simpler in the scope of Trezor because you don't have Carano App installed on it and everything works out of the box. So, you will always communicate with the last physically connected device.

Reply:
From user's point of view this is not simpler, because as I understand you have to call `initializeTrezorTransport` separately, as in [this test](https://github.com/input-output-hk/cardano-js-sdk/pull/238/files#diff-7f5c8ec02d8adf78f37dfc3c9f2ef1161ae277a22ea00a9370d9209954fc4057R31) - for both creating the key agent, and restoring the key agent. It would be great to encapsulate this to be called within `LedgerKeyAgent`. I think it can be done if you add required props to `createWithDevice` and serializableData :thinking:

Reply:
@mkazlauskas but `initializeTrezorTransport` must be called only once and it won't return deviceConnection but just prepare the app to accept signals from "transport". It should be called ideally on the app start.

Reply:
I understand, but why can't it be called internally?

Reply:
> initializeTrezorTransport must be called only once

Doesn't `createWithDevice` only get called once too? If so, is there a reason the transport cannot be initialised there?

Reply:
@mkazlauskas it could be called internally but then we need to know if is already initialized or not since it will throw an error if is called multiple times. From my perspective that is more complicated for users because they will need to take care of whether TrezorConnect has been already initialized or not. That is why I decided to do that in this way 🤔 . So, one initialization responsible for multiple devices and multiple trezor key agents and you don't user doesn't need to care if some key agent already exists to check that the connection is already initialized and to pass a prop to say to new key agent that is already initialized

@rhyslbw `Doesn't createWithDevice only get called once too? `:
yeah, it is called once per keyAgent but TrezorConnect should be initialized once on app start for all agents not per device

Maybe I'm wrong and don't see something 🤔  Don't get me wrong please, I just want to explain why I did that on this way :)

Reply:
> then we need to know if is already initialized or not since it will throw an error if is called multiple times

Don't see this as a problem - we can keep track if it was already initialized with a module-scoped variable, and throw if it's already initialized but we receive different props (different communication type or different manifest).

Reply:
Given the goal is to make the key agent implementations as common as possible, it would be best to try and keep the concerns of initialising the global Trezor transport internal.
 
> then we need to know if is already initialised or not since it will throw an error if is called multiple times

How about converting this logic to being an idempotent operation. i.e catch and inspect the error, returning `void` if it's the expected error, or re-throw if not.

Reply:
ok, I will change that